test/cpp/rpc/e2e_test_base.h

```diff
@@ -45,6 +47,12 @@ class TestE2EBase : public ::testing::Test {
     rpcAgent->start();
   }
 
+  void TearDown() override {
+    rpcAgent->join();
+    rpcAgent->shutdown();
+    RpcAgent::setCurrentRpcAgent(nullptr);
```

Comment:
Curious, do we have to do this, or just good habit?

Reply:
We do, because if we build the two tests (for the two agents) in a single binary then they are run one after the other, and if the first test doesn't unset the global agent then the second test will get an exception when trying to re-set it to a new value. (Because we have a check for that and explicitly raise an exception).